kernel/src/main/java/org/kframework/kompile/DefinitionParsing.java

```diff
@@ -269,69 +271,76 @@ protected Definition resolveConfigBubbles(Definition definition, Module defaultC
     }
 
     private Definition resolveConfigBubbles(Definition def) {
-      RuleGrammarGenerator gen = new RuleGrammarGenerator(def);
-      return DefinitionTransformer.from(m -> resolveConfigBubbles(def, m, gen), "parsing configs").apply(def);
-    }
-
-    private Module resolveConfigBubbles(Definition def, Module inputModule, RuleGrammarGenerator gen) {
-        if (stream(inputModule.localSentences())
-                .noneMatch(s -> s instanceof Bubble && ((Bubble) s).sentenceType().equals(configuration)))
-            return inputModule;
-
-        Set<Sentence> importedConfigurationSortsSubsortedToCell = stream(inputModule.productions())
-                .filter(p -> p.att().contains("cell"))
-                .map(p -> Production(Seq(), Sorts.Cell(), Seq(NonTerminal(p.sort())))).collect(Collections.toSet());
-
-        Module module = Module(inputModule.name(), (Set<Module>) inputModule.imports(),
-                (Set<Sentence>) inputModule.localSentences().$bar(importedConfigurationSortsSubsortedToCell),
-                inputModule.att());
+        Definition defWithCaches = resolveCachedBubbles(def, false);
+        RuleGrammarGenerator gen = new RuleGrammarGenerator(def);
+
+        Definition defWithParsedConfigs = DefinitionTransformer.from(m -> {
+            if (stream(m.localSentences()).noneMatch(s -> s instanceof Bubble && ((Bubble) s).sentenceType().equals(configuration)))
+                return m;
+            Module ruleParserModule = gen.getConfigGrammar(m);
+
+            ParseCache cache = loadCache(ruleParserModule);
+            try (ParseInModule parser = RuleGrammarGenerator.getCombinedGrammar(cache.getModule(), isStrict, profileRules, files)) {
```

Comment:
Here is the first ParseInModule being created. It gets `initialize` and `getScanner` called on it so that the object is fully computed prior to the split into multiple threads, which is important, otherwise the work gets done once on each thread.